settlement-pipelines/src/bin/init_settlement.rs

```diff
@@ -0,0 +1,747 @@
+            || merkle_tree_collection.slot != settlement_collection.slot
+        {
+            Err(anyhow!(
+            "Mismatched merkle tree and settlement collections: [array len: {} vs {}, epoch: {} vs {}, slot: {} vs {}]",
+            merkle_tree_collection.merkle_trees.len(),
+            settlement_collection.settlements.len(),
+            merkle_tree_collection.epoch, settlement_collection.epoch, merkle_tree_collection.slot, settlement_collection.slot
+        ))
+        } else {
+            Ok(CombinedMerkleTreeSettlementCollections {
+                slot: settlement_collection.slot,
+                epoch: settlement_collection.epoch,
+                merkle_tree_settlements: merkle_tree_collection
+                    .merkle_trees
+                    .into_iter()
+                    .zip(settlement_collection.settlements.into_iter())
+                    .map(|(merkle_tree, settlement)| MerkleTreeMetaSettlement {
+                        merkle_tree,
+                        settlement,
+                    })
+                    .collect(),
+            })
+        }
+    }?;
+
+    // Initialize the Anchor Solana client
+    let rpc_url = args.global_opts.rpc_url.expect("RPC URL is required");
+    let anchor_cluster = Cluster::from_str(&rpc_url)
+        .map_err(|e| anyhow!("Could not parse JSON RPC url `{:?}`: {}", rpc_url, e))?;
+
+    let default_keypair = load_default_keypair(args.global_opts.keypair.as_deref())?;
+    let fee_payer_keypair = if let Some(fee_payer) = args.global_opts.fee_payer {
+        load_keypair(&fee_payer)?
+    } else {
+        default_keypair.clone().map_or(Err(anyhow!("Neither --fee-payer nor --keypair provided, no keypair to pay for transaction fees")), Ok)?
+    };
+    let operator_authority_keypair = if let Some(operator_authority) = args.operator_authority {
+        load_keypair(&operator_authority)?
+    } else {
+        default_keypair.map_or(
+            Err(anyhow!(
+                "Neither --operator-authority nor --keypair provided, operator keypair required"
+            )),
+            Ok,
+        )?
+    };
+
+    let epoch = args.epoch.map_or_else(|| {
+        let merkle_file_name = Path::new(&args.input_merkle_tree_collection).file_name().
+            ok_or(anyhow!("Cannot extract file name from input merkle tree collection file path '{}'", args.input_merkle_tree_collection))?
+            .to_str().ok_or(anyhow!("Cannot convert file name {} to string", args.input_merkle_tree_collection))?;
+        let re = Regex::new("^([0-9]+)").unwrap();
+        let captures = re.captures(merkle_file_name);
+        captures
+            .and_then(|c| c.get(0))
+            .and_then(|m| m.as_str().parse::<u64>().ok())
+            .ok_or(
+                anyhow!("--epoch not provided and cannot extract epoch number from input merkle tree collection file path '{}'", args.input_merkle_tree_collection)
+            )
+    }, Ok)?;
```

Comment:
Not needed as the Collection has epoch inside (but let this be a note for the future iteration, not to change now)

Reply:
I realized that. Thanks for pointing it out